File: packages/lotos-dotnet/Contracts/ComponentContractRegistry.cs
namespace LotosUi.Dotnet.Contracts;

public static class ComponentContractRegistry
{
    private static readonly IReadOnlyDictionary<string, IReadOnlyCollection<string>> Contracts =
        new Dictionary<string, IReadOnlyCollection<string>>(StringComparer.OrdinalIgnoreCase)
        {
            ["button"] = new[] { "label", "variant", "size", "disabled" },
            ["badge"] = new[] { "label", "variant" },
            ["card"] = new[] { "title", "subtitle", "body", "footer" },
            ["input"] = new[] { "label", "placeholder", "helperText", "errorText" },
            ["form"] = new[] { "title", "description", "action", "method", "submitLabel", "body" },
            ["modal"] = new[] { "title", "description", "body", "footer" },
            ["table"] = new[] { "caption", "columns", "rows", "emptyState" },
        };

    public static bool HasComponent(string name) => Contracts.ContainsKey(name);

    public static IReadOnlyCollection<string> AllowedProps(string name) =>
        Contracts.TryGetValue(name, out var allowed) ? allowed : Array.Empty<string>();
}

File: packages/lotos-spring/src/main/java/com/lotosui/spring/contracts/ComponentContractRegistry.java
package com.lotosui.spring.contracts;

import java.util.List;
import java.util.Map;

public final class ComponentContractRegistry {
  private static final Map<String, List<String>> CONTRACTS = Map.of(
      "button", List.of("variant", "size", "disabled", "label"),
      "badge", List.of("variant", "label"),
      "card", List.of("title", "subtitle", "body", "footer"),
      "input", List.of("label", "type", "placeholder", "helperText", "errorText"),
      "form", List.of("title", "description", "method", "action", "submitLabel", "body"),
      "modal", List.of("title", "description", "open", "body", "footer"),
      "table", List.of("caption", "columns", "rows", "emptyState")
  );

  private ComponentContractRegistry() {
  }

  public static List<String> getAllowedProps(String component) {
    return CONTRACTS.getOrDefault(component, List.of());
  }

  public static boolean hasComponent(String component) {
    return CONTRACTS.containsKey(component);
  }
}
